internal/provisioner/plain/metrics/metrics.go

```diff
@@ -0,0 +1,100 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sirupsen/logrus"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+const (
+	// metricsPath is the path that the provisioner exposes its metrics at.
+	metricsPath = "/metrics"
+
+	// metricsPort is the port that the provisioner exposes its metrics over http.
+	metricsPort = 8383
+)
+
+const (
+	nameLabel      = "Name"
+	namespaceLabel = "Namespace"
+)
+
+var (
+	bundleGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "bundle",
+			Help: "Bundle Infomation. A positive value represents successful bundle unpack. A zero value represents bundle unpacking is pending. A negative value represents bundle unpacking has failed.",
+		},
+		[]string{nameLabel, namespaceLabel},
+	)
+)
+
+// ServeMetrics enables the provisioner to serve prometheus metrics.
+func ServeMetrics() error {
```

Comment:
Rather than running a separate metrics server, we should register these metrics with the existing controller-runtime metrics server.

https://book.kubebuilder.io/reference/metrics.html#publishing-additional-metrics

Reply:
Yup +1 on using anything we can use that we get for free.

Reply:
Reasoning for a custom server explained here: https://github.com/operator-framework/rukpak/issues/266#issuecomment-1111598474